src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1591 @@
+                public Derived(object o)
+                {
+                }
+                public Derived(string s)
+                {
+                }
+            }
+            """;
+
+        var comp2 = CreateCompilation([source2, OverloadResolutionPriorityAttributeDefinition], references: [comp1_1.ToMetadataReference()]);
+        comp2.VerifyDiagnostics();
+
+        var source3 = """
+            var c = new Derived("test");
+            """;
+
+        var comp3 = CreateCompilation(source3, references: [comp2.ToMetadataReference(), comp1_2.ToMetadataReference()]);
+        comp3.VerifyDiagnostics();
+
+        var derived = comp3.GetTypeByMetadataName("Derived")!;
+        var constructors = derived.Constructors;
+        Assert.Equal(2, constructors.Length);
+        Assert.All(constructors, m => Assert.IsType<RetargetingMethodSymbol>(m));
+
+        var tree = comp3.SyntaxTrees[0];
+        var model = comp3.GetSemanticModel(tree);
+        var creation = tree.GetRoot().DescendantNodes().OfType<ObjectCreationExpressionSyntax>().Single();
+        AssertEx.Equal("Derived..ctor(System.Object o)", model.GetSymbolInfo(creation).Symbol!.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void LangVersion()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                public void M(object o) => System.Console.Write("1");
+                public void M(string s) => throw null;
+
+                [OverloadResolutionPriority(1)]
+                public int this[object o]
+                {
+                    get
+                    {
+                        System.Console.Write("2");
+                        return 1;
+                    }
+                    set
+                    {
+                        System.Console.Write("3");
+                    }
+                }
+                public int this[string s]
+                {
+                    get => throw null;
+                    set => throw null;
+                }
+
+                [OverloadResolutionPriority(1)]
+                public C(object o)
+                {
+                    System.Console.Write("4");
+                }
+
+                public C(string s)
+                {
+                    throw null;
+                }
+            }
+            """;
+
+        CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition], parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (5,6): error CS8652: The feature 'overload resolution priority' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "OverloadResolutionPriority(1)").WithArguments("overload resolution priority").WithLocation(5, 6),
```

Comment:
>ERR_FeatureInPreview [](http://example.com/codeflow?start=33&length=20)

Is it necessary to report an error where the attribute is applied, rather than at the call-site?

Reply:
Yes. We don't report errors at the call site, even when the lang version is lower. See the last line of this test.